developers/utils/tools.md

```diff
@@ -5,9 +5,207 @@ title: Framework Tools
 
 # Framework Tools
 
-Find a few framework utilities listed in this chapter.
+The openHAB framework provides a number of Java tools and utilities so that addon developers do not have to _"re-invent the wheel"_ when writing their own addon code:
 
-# Unique Instance ID
+- [Channel Types](#channel-types)
+- [Thing Properties](#thing-properties)
+- [Hex Utilities](#hex-utilities)
+- [Statistics Utilities](#statistics-utilities)
+- [String Utilities](#string-utilities)
+- [Lighting Model and Color Utilities](#lighting-model-and-color-utilities)
+- [Unique ID Utilities](#unique-id-utilities)
+- [Network Configuration Utilities](#network-address-service)
+- [Caching Support](#caching)
+
+## Channel Types
+
+The openHAB framework provides many standard channel types.
+These should be used wherever possible in preference to defining your own.
+
+```java
+    ChannelType on = DefaultSystemChannelTypeProvider.SYSTEM_POWER;
+```
+
+## Thing Properties
+
+The openHAB framework provides many standard property types.
+These should be used wherever possible in preference to defining your own.
+
```

Comment:
What advantages offer the standard property types, like `Thing.PROPERTY_SERIAL_NUMBER`?  A disadvantage is that the property names are not translated depending on the Regional Settings or Browsers preferences.

Reply:
> Properties are not supposed to be translated. In that way they are similar to channel ids. 

Channel ids are fixed, but the description of the channels can be translated, I do not think this is similar.  The user is supposed to see the property names and property values (like “Has Bluetooth module: YES”), so in my eyes the property names should be translated.  Theoretically this will not prevent the creation of a list of firmware versions of all devices in openHAB, just the user should see instead of “Firmware“ or “Serial Number” as property name translated text.

Besides in the openHAB documentation I am not aware of text discouraging the internationalization/localization of property names and values.

Reply:
Properties should be camelCase, this is documented here: https://www.openhab.org/docs/developer/bindings/thing-xml.html#properties

From a technical discovery perspective, properties and configuration parameters are both provided as properties in camelCase and can also be defined statically in Thing XML definition. The I18N tool does not make these available for translation.

As @andrewfg proposed, please create an issue for continuing this discussion.